src/components/Breadcrumbs/BreadcrumbsItem.tsx

```diff
@@ -3,21 +3,23 @@ import {Link} from '../Link';
 import {block} from '../utils/cn';
 import type {BreadcrumbsItem as IBreadcrumbsItem, BreadcrumbsProps} from './Breadcrumbs';
 
-interface Props {
-    data: IBreadcrumbsItem;
+interface Props<T extends IBreadcrumbsItem = IBreadcrumbsItem> {
+    data: T;
     isCurrent: boolean;
     isPrevCurrent: boolean;
-    renderItem?: BreadcrumbsProps['renderItemContent'] | BreadcrumbsProps['renderRootContent'];
+    renderItem?:
+        | BreadcrumbsProps<T>['renderItemContent']
+        | BreadcrumbsProps<T>['renderRootContent'];
 }
 
 const b = block('breadcrumbs');
 
-export const BreadcrumbsItem = React.memo(function BreadcrumbsItem({
+function BreadcrumbsItem<T extends IBreadcrumbsItem = IBreadcrumbsItem>({
     data,
     isCurrent,
     isPrevCurrent,
     renderItem,
```

Comment:
Isn't it enough to make function generic?

```suggestion
export const BreadcrumbsItem = React.memo(function BreadcrumbsItem<T extends IBreadcrumbsItem = IBreadcrumbsItem>({
    data,
    isCurrent,
    isPrevCurrent,
    renderItem,
```

Reply:
Yes, it is. But the last version has less changed rows. But if you insist I will bring back the version with function.

Reply:
There is also `displayName` disappear, and i personally like named instead of `default`

Reply:
Please suggest your fix for the file.

Reply:
I suggest (above) to just add generic to function and keep everything other unchanged

```jsx
export const BreadcrumbsItem = React.memo(function BreadcrumbsItem<T extends IBreadcrumbsItem = IBreadcrumbsItem>({
    data,
    isCurrent,
    isPrevCurrent,
    renderItem,
}: Props<T>) {
    //
});

BreadcrumbsItem.displayName = 'Breadcrumbs.Item';
```

Reply:
Please provide full solution, your one has no any generic-component, and there is nothing you can use for type-cast as described in link above.